sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/dto/PlanRegistrationRequest.java

```diff
@@ -0,0 +1,22 @@
+package com.kyriba.conference.sponsorship.api.dto;
+
+import com.kyriba.conference.sponsorship.domain.PlanCategory;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
+/**
+ * @author M-ASL
+ * @since v1.0
+ */
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class PlanRegistrationRequest
+{
+  private PlanCategory category;
+  private String sponsorId;

```

Comment:
Why does plan have concrete sponsor?

Reply:
Lets suppose, there are several sponsorship plans available. When some sponsor chooses a plan and pays for it, then the plan becomes reserved for this sponsor. All sponsorship plans with an empty sponsorID are available for reservation. One sponsor can buy several plans. So it is one-to-many relationship and it is logical to have reference to a sponsor from a plan.